### PR TITLE
Filter, Parser: discarded, matched, not matched messages statistics

### PR DESCRIPTION
**Pull request by furiel**
Discarded statistics counter for parsers.
Matched, not matched statistics counters for filters.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests passed on test branch: master and test suite: functions

File: lib/filter/tests/test_filters_statistics.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <glib.h>
+
+#include "testutils.h"
```
Comment (kvch) [inline lib/filter/tests/test_filters_statistics.c]: Please write new unittests using Criterion.

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: lib/logpipe.h [hunk 2]
```diff
@@ -213,6 +213,7 @@ struct _LogPipe
   GlobalConfig *cfg;
   LogExprNode *expr_node;
   LogPipe *pipe_next;
+  StatsCounterItem *stats_discarded;
```
Comment (kvch) [inline lib/logpipe.h]: I would rename it to `discarded_messages`, to stay consistent with other counters' names.

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: lib/stats/stats-cluster.c [hunk 3]
```diff
@@ -119,6 +119,8 @@ _get_module_name(gint source)
     "java",
     "http",
     "python"
```
Comment (furiel) [inline lib/stats/stats-cluster.c]: after rebase the comma went missing. @furiel please fix before merge

**Comment by kira-syslogng** [comment]
Build FAILURE, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build FAILURE, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build FAILURE, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by pzoleex** [comment]
@kira-syslogng test this please test branch=pzolee-stats-filter-parser-discarded;

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: pzolee-stats-filter-parser-discarded and test suite: functions

**Comment by pzoleex** [comment]
from test side, this pr is ok

**Pull request merged**